packages/babel-parser/test/fixtures/experimental/export-extensions/default-type-with-typescript/options.json

```diff
@@ -0,0 +1,5 @@
+{
+  "plugins": ["exportDefaultFrom", "typescript"],
+  "sourceType": "module",
+  "throws": "Unexpected token, expected \"=\" (1:17)"
```

Comment:
Did this work before this PR?

Reply:
It throws before this PR because `isExportDefaultSpecifier` returns false

https://github.com/babel/babel/blob/865d5155c2ec91f1122e94b88e04ee902dfd6677/packages/babel-parser/src/plugins/typescript/index.js#L1614

It may be copied and revised from `flow` plugin. I think we should consider relax the constraint unless banned by the `export-default-from` proposal.